signature-envelope-cose.md

```diff
@@ -0,0 +1,197 @@
+# COSE Sign1 Signature Envelope
+
+This specification implements the [Notary v2 Signature specification](signature-specification.md) using 
+CBOR Object Signing and Encryption (COSE). COSE ([RFC8152](https://datatracker.ietf.org/doc/html/rfc8152)) is a CBOR based envelope format for digital signatures over any type of payload (e.g. CBOR, JSON, binary).
+Notary v2 specifically supports [COSE_Sign1_Tagged](https://datatracker.ietf.org/doc/html/rfc8152#section-4.2) as a signature envelope.
+
+## Storage
+
+A COSE signature envelope will be stored in an OCI registry as a blob, and referenced in the signature manifest as a blob with `mediaType` of `"application/cose"`.
```

Comment:
a more precise media type to use in this context would be `application/cose; cose-type="cose-sign1"` (see [Section 2 of STD 96](https://www.rfc-editor.org/rfc/rfc9052#section-2)).

Reply:
I think this is redundant since the COSE envelopes are tagged anyway. In my mind this would make client code harder to write because now you need to define how media type parameters are encoded in this context. This is not HTTP (which has its own definition for encoding those).

Reply:
> I think this is redundant since the COSE envelopes are tagged anyway.

Yeah it's redundant.  Whether this hint is worthwhile depends on how processing is implemented.  With the extra information, the COSE payload can be dispatched directly to the right API, whilst without, the implementation needs to peek into the COSE tag to dispatch.

> In my mind this would make client code harder to write because now you need to define how media type parameters are encoded in this context. This is not HTTP (which has its own definition for encoding those).

Not sure I understand what the encoding problem is?  The media parameters that'd be used here are all UTF-8 strings (which both JSON and I-JSON support off-the-shelf) with a-priori known values.

( I have the feeling I am missing something though :-) )

Reply:
In practice, even outside the context of Email or HTTP headers, people use the same syntax as first defined in https://datatracker.ietf.org/doc/html/rfc2045#section-5.1 to encode parameters, especially for protocols/formats where you want to use a single string to encode the media type. 
But it has to be made explicit with a reference that this is what's happening here.
From an implementation point of view, for example in Python, clients would have to do the following to match the media type (note the `parse_header` function name):
```py
import cgi
mediatype, params = cgi.parse_header('application/cose; cose-type="cose-sign1"')
if mediatype == "application/cose" and params["cose-type"] == "cose-sign1":
  # process
```
This is necessary since whitespace becomes a factor now. Previously, a simple string comparison would have been enough: `mediatype == "application/cose"`.
Of course you could prescribe that the media type string must always use a specific fixed string that can be matched by string comparison. There is some chance that people will mess it up though.

I think, given that the ORAS artifact type already says that it is a (single) signature (`"artifactType": "application/vnd.cncf.notary.signature"`), I don't see why it has to be repeated here using the parameter.

Reply:
Agree with letmaik - going to close this comment out.